### PR TITLE
feat(tooltips): new light/dark mode colors

### PR DESCRIPTION
**Pull request by ze-flo**
## Description

Adds light / dark mode support for Tooltips

## Detail

Screenshot 2024-05-23 at 2 00 13 PM
Screenshot 2024-05-23 at 2 00 36 PM
Screenshot 2024-05-23 at 2 01 35 PM
Screenshot 2024-05-23 at 2 01 54 PM

## Checklist

- [ ] :ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- [ ] ~~:arrow_left: renders as expected with reversed (RTL) direction~~
- [ ] ~~:metal: renders as expected with Bedrock CSS (`?bedrock`)~~
- [x] :guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)
- [x] :wheelchair: tested for WCAG 2.1 AA accessibility compliance
- [x] :memo: tested in Chrome, Firefox, Safari, and Edge

### CODE REVIEW

**Comment by ze-flo** [comment]
The drop-shadow casts a shadow on the tooltip's arrow tip. This is particularly visible in Dark mode.

Screenshot 2024-05-23 at 2 18 39 PM

The problem should automatically be resolved once either https://github.com/zendeskgarden/react-components/pull/1812 or https://github.com/zendeskgarden/react-components/pull/1812 is merged.

File: packages/tooltips/src/styled/StyledTooltip.ts [hunk 1]
```diff
@@ -121,30 +121,49 @@ const sizeStyles = ({
...
+
+  let backgroundColor = getColor({
+    theme,
+    hue: 'neutralHue',
+    light: { shade: 900 },
+    dark: { shade: 700 }
+  });
+
+  let borderColor = backgroundColor;
+  let color = theme.palette.white;
```
Comment (jzempel) [inline packages/tooltips/src/styled/StyledTooltip.ts:144]:
```suggestion
  let color = getColor({ hue: 'white' });
```

Ensure a best possible future-proof lookup.

File: packages/tooltips/src/styled/StyledTooltip.ts [hunk 2]
```diff
@@ -121,30 +121,49 @@ const sizeStyles = ({
...
-  let color = getColorV8('background', 600 /* default shade */, theme);
+
+  let backgroundColor = getColor({
+    theme,
+    hue: 'neutralHue',
+    light: { shade: 900 },
+    dark: { shade: 700 }
+  });
+
+  let borderColor = backgroundColor;
```
Comment (jzempel) [inline packages/tooltips/src/styled/StyledTooltip.ts:143]:
```suggestion
  let borderColor = 'transparent';
```

[nit] this is Garden's usual approach for retaining border size without styling.

File: packages/tooltips/src/styled/StyledTooltip.ts [hunk 3]
```diff
@@ -121,30 +121,49 @@ const sizeStyles = ({
...
   );
-  let backgroundColor = getColorV8('chromeHue', 700, theme);
-  let color = getColorV8('background', 600 /* default shade */, theme);
+
+  let backgroundColor = getColor({
+    theme,
+    hue: 'neutralHue',
+    light: { shade: 900 },
+    dark: { shade: 700 }
+  });
```
Comment (jzempel) [inline packages/tooltips/src/styled/StyledTooltip.ts:141]: Feels like all this assignment is premature and could be micro-optimized by moving to an `else` block below `if (type === 'light')`. Also ends up being slightly more readable.

**Review by jzempel — APPROVED** [review]
Pending design approval

**Comment by ze-flo** [comment]
> The drop-shadow casts a shadow on the tooltip's arrow tip. This is particularly visible in Dark mode.
> 
> Screenshot 2024-05-23 at 2 18 39 PM
> 
> The problem should automatically be resolved once either #1812 or #1814 is merged.
---

> [!NOTE]
> https://github.com/zendeskgarden/react-components/pull/1814 fixed the drop-shadow arrow issue. 🥳 💯 

Screenshot 2024-05-30 at 12 50 35 PM

**Review by sieeeeeenna — APPROVED** [review]
Everything looks good to me! The only thing I noticed is the small light tooltip has a larger shadow and the fg/subtle text instead of fg/default (see screenshot for the Figma version). I don't believe the small light tooltip is ever used and Garden doesn't recommend it anyway, so if that it doesn't matter then disregard. Other than that, looks good!!! :)
Screenshot 2024-05-30 at 4 24 10 PM

**Comment by ze-flo** [comment]
> Everything looks good to me! The only thing I noticed is the small light tooltip has a larger shadow and the fg/subtle text instead of fg/default (see screenshot for the Figma version). I don't believe the small light tooltip is ever used and Garden doesn't recommend it anyway, so if that it doesn't matter then disregard. Other than that, looks good!!! :) Screenshot 2024-05-30 at 4 24 10 PM

@sieeeeeenna You are spot-on for calling this out. 💯 The component's API isn't as restrictive and allows for combinations design doesn't support. But I think it's safe to assume the use case you described will never happen IRL.

**Pull request merged**